catalyst-gateway/bin/src/db/index/index_cip36.rs

```diff
@@ -0,0 +1,243 @@
+
+        insert_queries
+    }
+}
+
+/// Insert CIP-36 Registration Queries
+pub(crate) struct Cip36InsertQuery {
+    /// Stake Registration Data captured during indexing.
+    cip36_reg_data: Vec<Cip36RegistrationInsertQuery>,
+}
+
+impl Cip36InsertQuery {
+    /// Create new data set for CIP-36 Registrations Insert Query Batch.
+    #[allow(dead_code)]
+    pub(crate) fn new() -> Self {
+        Cip36InsertQuery {
+            cip36_reg_data: Vec::new(),
+        }
+    }
+
+    /// Prepare Batch of Insert CIP-36 Registrations Index Data Queries
+    pub(crate) async fn prepare_batch(
+        session: &Arc<Session>, cfg: &CassandraEnvVars,
+    ) -> anyhow::Result<SizedBatch> {
+        Cip36RegistrationInsertQuery::prepare_batch(session, cfg).await
+    }
+
+    /// Prepare Batch of Insert CIP-36 Registrations Index Data Queries
+    pub(crate) async fn prepare_batch_by_vote_key(
+        session: &Arc<Session>, cfg: &CassandraEnvVars,
+    ) -> anyhow::Result<SizedBatch> {
+        Cip36RegistrationInsertQuery::prepare_batch_by_vote_key(session, cfg).await
+    }
+
+    /// Index the CIP-36 registrations in a transaction.
+    #[allow(dead_code)]
+    pub(crate) fn index(
+        &mut self, txn: usize, txn_index: i16, slot_no: u64, block: &MultiEraBlock,
+    ) {
+        if let Some(decoded_metadata) = block.txn_metadata(txn, Metadata::cip36::LABEL) {
+            #[allow(irrefutable_let_patterns)]
+            if let Metadata::DecodedMetadataValues::Cip36(cip36) = &decoded_metadata.value {
+                for vote_key in &cip36.voting_keys {
+                    let vote_key = vote_key.voting_pk.to_bytes().to_vec();
+                    let stake_address = cip36.stake_pk.map(|s| s.to_bytes().to_vec());
+                    let payment_address = Some(cip36.payment_addr.clone());
+                    let is_payable = Some(cip36.payable);
+                    let raw_nonce = Some(cip36.raw_nonce);
+                    let nonce = Some(cip36.nonce);
+                    let purpose = Some(cip36.purpose);
+                    let signed = Some(cip36.signed);
+                    let strict_catalyst = cip36.cip36;
+                    let error_report = Some(decoded_metadata.report.clone());
+                    self.cip36_reg_data.push(Cip36RegistrationInsertQuery::new(
+                        stake_address,
+                        vote_key,
+                        slot_no,
+                        txn_index,
+                        payment_address,
+                        is_payable,
+                        raw_nonce,
+                        nonce,
+                        purpose,
+                        signed,
+                        strict_catalyst,
+                        error_report,
+                    ));
+                }
+            }
+        }
+    }
+
+    /// Execute the CIP-36 Registration Indexing Queries.
+    ///
+    /// Consumes the `self` and returns a vector of futures.
+    #[allow(dead_code)]
```

Comment:
probably redundant `#[allow(dead_code)]`

Reply:
Thanks for the catch!